### DISCUSSION TITLE
Detect `functools.cached_property` usage with `dataclass(slots=True)`

### DISCUSSION THREAD

**Discussion by minmax** [astral-sh/ruff#10940] [Ideas]
I'd like to propose a new rule for linter, catch `functools.cached_property` used with `dataclass(slots=True)`.

```
from dataclasses import dataclass
from functools import cached_property

@dataclass(slots=True)
class F:
    @cached_property
    def prop(self) -> int:
        return 1

print(F().prop)
```

This code fails in runtime with:

```
TypeError: No '__dict__' attribute on 'F' instance to cache 'prop' property.
```

If `slots=True` was used, then the usage of `functools.cached_property` should be suppressed.

**Comment by MichaReiser** [comment]
I assume the rule should cover all cases where a class uses slots.
